advanced_animation/core/storyboard_generator.py

```
@@ -270,6 +245,185 @@ def _generate_ai_storyboard(self, code_analysis: Dict[str, Any]) -> Storyboard:
+                )
+                response_content = response.choices[0].message.content
+            
+            # Parse the response
+            if response_content:
+                return json.loads(response_content)
+            else:
+                logger.error("Empty response from AI provider")
+                return None
+                
+        except Exception as e:
+            logger.error(f"Error calling AI API: {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)